internal/server/internal/ecdsa/ecdsa.go

```diff
@@ -0,0 +1,308 @@
+package ecdsa
+
+import (
+	"crypto"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"encoding/asn1"
+	"errors"
+	"io"
+	"math/big"
+)
+
+type rand struct {
+	kInv, r *big.Int
+}
+
+// NOTE: This buffer is designed to ensure that, regardless of how code outside
+// of this package uses these types and methods, no random value will ever be
+// used more than once in an ECDSA signature. When modifying this code, be VERY
+// CAREFUL that this behavior is maintained.
+
+// A RandProvider is capable of producing random values for use in ECDSA
+// signatures. It cannot be implemented by types outisde of this package.
+type RandProvider interface {
```

Comment:
Search for `buf_ecdsa` in `server.go` - I had to rename the import so it wouldn't conflict with Go's stdlib `ecdsa` package.